backend/src/v1/services/analytic-service.ts

```diff
@@ -0,0 +1,66 @@
+import { config } from '../../config';
+import {
+  PowerBiService,
+  ReportInWorkspace,
+} from '../../external/services/powerbi-service';
+
+// Embed info for Reports, Dashboards, and other resources
+export type PowerBiEmbedInfo = {
+  resources: { name: PowerBiResourceName; id: string; embedUrl: string }[];
+  accessToken: string;
+  expiry: string;
+};
+
+export enum PowerBiResourceName {
+  SubmissionAnalytics = 'SubmissionAnalytics',
+  UserBehaviour = 'UserBehaviour',
+  DataAnalytics = 'DataAnalytics',
+}
+
+const resourceIds: Record<PowerBiResourceName, ReportInWorkspace> = {
+  SubmissionAnalytics: {
+    workspaceId: config.get('powerbi:analytics:workspaceId'),
+    reportId: config.get('powerbi:analytics:submissionAnalyticsId'),
+  },
+  UserBehaviour: {
+    workspaceId: config.get('powerbi:analytics:workspaceId'),
+    reportId: config.get('powerbi:analytics:userBehaviourId'),
+  },
+  DataAnalytics: {
+    workspaceId: config.get('powerbi:analytics:workspaceId'),
+    reportId: config.get('powerbi:analytics:dataAnalyticsId'),
+  },
+};
+
+/**
+ * Generate embed token and embed urls for PowerBi resources
+ * @return Details like Embed URL, Access token and Expiry
+ */
+export async function getEmbedInfo(
+  resourceNames: PowerBiResourceName[],
+): Promise<PowerBiEmbedInfo> {
+  const powerBi = new PowerBiService(
+    config.get('entra:clientId'),
+    config.get('entra:clientSecret'),
+    config.get('entra:tenantId'),
+  );
+
+  const embedParams = await powerBi.getEmbedParamsForReports(
+    resourceNames.map((name) => resourceIds[name]),
+  );
+
+  const resources = [];
+  for (let i = 0; i < resourceNames.length; ++i) {
```

Comment:
I've added a zod verification to ensure that it is an array. I've also added a test case to show that if it's not an array it will return 500. Is this enough to dismiss this alert? I feel anything else I do would be redundant.